include/deal.II/base/vectorization.h

```diff
@@ -379,7 +381,7 @@ class VectorizedArray
    * Actual data field. Since this class represents a POD data type, it is
    * declared public.
    */
-  Number data;
+  Number data = 0.;
```

Comment:
Can you explain why you need this change (and for the various specializations below)? This is a fundamental change in behavior:
- It is a performance regression because I don't think the compiler will be allowed to skip initialization of `VectorizedArray` to zero when e.g. creating a fixed-size array on the stack.
- This makes this class behave differently than the basic types `double` because a plain variable is initialized, which is undesired.

Shouldn't everything you need be covered by the non-default constructor that you added?

Reply:
Initially I added these member initialization so that I can remove the default constructor, but we added back all of the constructors and these direct member initialization are not needed.

I was not aware of the performance implications of this. So now I removed member initialization in df218cb. The tests are passing on my machine.